apps/stripe/src/app/api/gateway-initialize/use-case.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  buildSyncWebhookResponsePayload,
+  SyncWebhookResponsesMap,
+} from "@saleor/app-sdk/handlers/shared";
+import { err, ok, Result } from "neverthrow";
+
+import {
+  PaymentGatewayInitializeResponseShape,
+  PaymentGatewayInitializeResponseShapeType,
+} from "@/app/api/gateway-initialize/response-shape";
+import { BaseError } from "@/lib/errors";
+import { AppConfigPersistor } from "@/modules/app-config/app-config-persistor";
+
+type UseCaseResultShape = SyncWebhookResponsesMap["PAYMENT_GATEWAY_INITIALIZE_SESSION"];
+
+type UseCaseErrorShape = InstanceType<typeof InitializeStripeSessionUseCase.UseCaseError>;
+
+export class InitializeStripeSessionUseCase {
+  private configPersistor: AppConfigPersistor;
+
+  static UseCaseError = BaseError.subclass("InitializeStripeSessionUseCaseError");
+  static MissingConfigError = this.UseCaseError.subclass("MissingConfigError");
+
+  constructor(deps: { configPersistor: AppConfigPersistor }) {
+    this.configPersistor = deps.configPersistor;
+  }
+
+  async execute(params: {
+    channelId: string;
+    appId: string;
+    saleorApiUrl: string;
+  }): Promise<Result<UseCaseResultShape, UseCaseErrorShape>> {
+    const { channelId, appId, saleorApiUrl } = params;
+
+    const stripeConfigForThisChannel = await this.configPersistor.getStripeConfig({
+      channelId,
+      appId,
+      saleorApiUrl,
+    });
+
+    if (stripeConfigForThisChannel.isOk()) {
+      const pk = stripeConfigForThisChannel.value?.getPublishableKey();
+
+      if (!pk) {
+        return err(
+          new InitializeStripeSessionUseCase.MissingConfigError("Config for channel not found", {
+            // todo: pass props (channel id), but type safe
+          }),
+        );
+      }
+
+      const rawShape: PaymentGatewayInitializeResponseShapeType = {
+        stripePublishableKey: pk.getKeyValue(),
+      };
+
+      const responseDataShape = PaymentGatewayInitializeResponseShape.parse(rawShape);
+
+      const validResponseShape =
+        buildSyncWebhookResponsePayload<"PAYMENT_GATEWAY_INITIALIZE_SESSION">({
+          data: responseDataShape,
+        });
+
+      return ok(validResponseShape);
+    }
+
+    if (stripeConfigForThisChannel.isErr()) {
+      return err(
+        new InitializeStripeSessionUseCase.UseCaseError(
+          "Failed to retrieve Publishable Key from config",
+          {
+            cause: stripeConfigForThisChannel.error,
+          },
+        ),
+      );
+    }
+
+    throw new Error("Leaky logic, should not happen");
```

Comment:
```suggestion
    throw new BaseError("Leaky logic, should not happen");
```
It will help us with error serialization etc